pkg/eventservice/event_broker_test.go

```diff
@@ -181,45 +180,98 @@ func TestCURDDispatcher(t *testing.T) {
 
 	dispInfo := newMockDispatcherInfoForTest(t)
 	// Case 1: Add and get a dispatcher.
-	broker.addDispatcher(dispInfo)
-	disp := broker.getDispatcher(dispInfo.GetID())
+	err := broker.addDispatcher(dispInfo)
+	require.Nil(t, err)
+	disp := broker.getDispatcher(dispInfo.GetID()).Load()
 	require.NotNil(t, disp)
 	require.Equal(t, disp.id, dispInfo.GetID())
 
 	// Case 2: Reset a dispatcher.
 	dispInfo.startTs = 1002
-	broker.resetDispatcher(dispInfo)
-	disp = broker.getDispatcher(dispInfo.GetID())
+	dispInfo.epoch = 2
+	err = broker.resetDispatcher(dispInfo)
+	require.Nil(t, err)
+	disp = broker.getDispatcher(dispInfo.GetID()).Load()
 	require.NotNil(t, disp)
 	require.Equal(t, disp.id, dispInfo.GetID())
 	// Check the resetTs is updated.
-	require.Equal(t, disp.resetTs.Load(), dispInfo.GetStartTs())
+	require.Equal(t, disp.startTs, dispInfo.GetStartTs())
 
 	// Case 3: Pause a dispatcher.
 	broker.pauseDispatcher(dispInfo)
-	disp = broker.getDispatcher(dispInfo.GetID())
+	disp = broker.getDispatcher(dispInfo.GetID()).Load()
 	require.NotNil(t, disp)
 	require.False(t, disp.isReadyReceivingData.Load())
 
 	// Case 4: Resume a dispatcher.
 	broker.resumeDispatcher(dispInfo)
-	disp = broker.getDispatcher(dispInfo.GetID())
+	disp = broker.getDispatcher(dispInfo.GetID()).Load()
 	require.NotNil(t, disp)
 	require.True(t, disp.isReadyReceivingData.Load())
 
 	// Case 5: Remove a dispatcher.
 	broker.removeDispatcher(dispInfo)
-	disp = broker.getDispatcher(dispInfo.GetID())
-	require.Nil(t, disp)
+	dispPtr := broker.getDispatcher(dispInfo.GetID())
+	require.Nil(t, dispPtr)
+}
+
+func TestResetDispatcher(t *testing.T) {
```

Comment:
Could you also add a test to ensure that if there are concurrent resets happening, only the largest epoch in dispatcherStat remains?